lib/rust/ensogl/core/src/display/camera/camera2d.rs

```diff
@@ -535,3 +543,37 @@ impl display::Object for Camera2d {
         &self.display_object
     }
 }
+
+
+
+// =============
+// === Tests ===
+// =============
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    /// A regression test checks whether handling the camera's FRP events does not cause panics
+    /// at runtime.
+    ///
+    /// If the events are emitted with the camera's internal [`RefCell`] lock held, any usage of
+    /// camera API in event handlers will cause a panic.
```

Comment:
Technically, not _any_ usage of camera API, but only those requiring mutable borrow (if they require only immutable borrow, and the lock is also immutable, the RefCell borrow is fine).

So make sure, that we test with the API requiring a mutable borrow.

Reply:
In a previous version of the code, the `position` event was emitted with the mutable lock held, so even an immutable borrow in some other method caused panic. My initial fix used immutable borrow, and it had an issue for methods with mutable borrow.  That's why we need to check both mutable and immutable borrows in the test. I will slightly rephrase this though